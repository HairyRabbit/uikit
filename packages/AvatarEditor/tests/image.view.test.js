/**
 * @jest
 */

import * as React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

/**
 * setup
 */

configure({ adapter: new Adapter() })

jest.mock('../style.css', () => {
  return {
    small: 'small',
    normal: 'normal',
    large: 'large',
    huge: 'huge',
    circle: 'circle',
    round: 'round',
    ratio: 2.25
  }
})


/**
 * src prop test
 */

test('should render image', () => {
  const Avatar = require('../image.view').default
  const wrapper = shallow(<Avatar src="foo" />)
  expect(wrapper.find('img').prop('src')).toEqual('foo')
})

test('should report error when not given src', () => {
  const mockfn = jest.fn()
  console.error = mockfn
  const Avatar = require('../image.view').default
  const wrapper = shallow(<Avatar />)
  expect(mockfn).toHaveBeenCalled()
})


/**
 * size prop test
 */

test('should use default size', () => {
  const Avatar = require('../image.view').default
  const wrapper = shallow(<Avatar src="foo" />)
  expect(wrapper.hasClass('normal')).toEqual(true)
})

test('should use build in size', () => {
  const Avatar = require('../image.view').default
  const wrapper = shallow(<Avatar size="small" src="foo" />)
  expect(wrapper.hasClass('small')).toEqual(true)
})

test('should custom size', () => {
  const Avatar = require('../image.view').default
  const wrapper = shallow(<Avatar size="1rem" src="foo" />)
  expect(wrapper.hasClass('normal')).not.toEqual(true)
  expect(wrapper.prop('style')).toEqual({
    width: '1rem',
    height: '1rem',
    lineHeight: '1rem',
    fontSize: 'calc(1rem / 2.25)'
  })
})

/**
 * snapshots test
 */

test('should match snapshot', () => {
  const Avatar = require('../image.view').default
  const tree = renderer
        .create(<Avatar src="foo" />)
        .toJSON()

  expect(tree).toMatchSnapshot()
})
