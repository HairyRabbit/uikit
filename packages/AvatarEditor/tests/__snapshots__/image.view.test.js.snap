// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className=" normal circle "
  style={Object {}}
>
  <img
    className={undefined}
    onError={undefined}
    src="foo"
  />
</div>
`;
